lib/models/device.coffee

```diff
@@ -106,7 +106,21 @@ getDeviceModel = (deps, opts) ->
 		ResourceNotFoundError: errors.BalenaDeviceNotFound
 	}
 
-	exports = {}
+	overallStatus = {
```

Comment:
How about moving this in a TS file and use an `enum` (not a `const enum`) and then import it both in here and in the `typings/balena-sdk.d.ts`?

Reply:
@thgreasi as far as I am aware, we cannot get something like `valueof` on an enum, so we will have duplication in the `overall_status` typings. I can still go for `enum` though, but we will have a duplication.

Reply:
Why not a `const enum` @thgreasi? And @sradevski the "valueof" type of an enum is just the enum itself

Reply:
@Page- I meant the union of all possible values for the enum, what `overalProgress[keyof typeof overallProgress]` achieves.

Reply:
Nevermind, I was looking at the code and I see what you mean, that would definitely make more sense, I was just thinking in `balena-request-status` terms.